assets/src/hooks/useMinischedule.ts

```diff
@@ -0,0 +1,20 @@
+import { useEffect, useState } from "react"
+import { fetchMinischeduleBlock, fetchMinischeduleRun } from "../api"
+import { Block, Run } from "../minischedule"
+import { TripId } from "../schedule"
+
+export const useMinischeduleRun = (tripId: TripId): Run | null => {
+  const [run, setRun] = useState<Run | null>(null)
+  useEffect(() => {
+    fetchMinischeduleRun(tripId).then(setRun)
+  }, [])
```

Comment:
Should this `useEffect` (and below) pass `tripId` as a dependency? If the vehicle's trip changes while this panel is open we'd want to load fresh data, right?

Reply:
Good catch!